modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -1701,6 +1706,40 @@ public void testGetObjectEntries() throws Exception {
 				"sort", "textObjectFieldName:desc"
 			).build(),
 			childObjectEntry2, childObjectEntry1);
+
+		// With Aggregation field and many_to_many relationship with himself
+
+		ObjectDefinition objectDefinition = _createObjectDefinition(
+			Arrays.asList(
+				new TextObjectFieldBuilder(
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).name(
+					"textObjectFieldName"
+				).build()));
+
+		ObjectRelationship objectRelationship =
+			_objectRelationshipLocalService.addObjectRelationship(
+				_adminUser.getUserId(),
+				objectDefinition.getObjectDefinitionId(),
+				objectDefinition.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"manyToManyRelationshipName",
+				ObjectRelationshipConstants.TYPE_MANY_TO_MANY);
+
+		_addAggregationObjectField(
+			"SUM", "id", objectRelationship.getName(),
+			objectDefinition.getObjectDefinitionId());
+
+		_assertObjectEntriesSize(0, objectDefinition);
+
+		_objectRelationshipLocalService.deleteObjectRelationship(
+			objectRelationship.getObjectRelationshipId());
+
+		_objectDefinitionLocalService.deleteObjectDefinition(
+			objectDefinition.getObjectDefinitionId());
```

Comment:
I think would be better if you insert this method in `testGetObjectEntriesAggregationFacets` method, since it is related to the aggregation context.
I suggest adding some entries to test if the size returns the correct value.